pkg/block/fetcher.go

```diff
@@ -469,15 +585,20 @@ type MetaFetcher struct {
 //
 // Returned error indicates a failure in fetching metadata. Returned meta can be assumed as correct, with some blocks missing.
 func (f *MetaFetcher) Fetch(ctx context.Context) (metas map[ulid.ULID]*metadata.Meta, partial map[ulid.ULID]error, err error) {
-	metas, partial, err = f.wrapped.fetch(ctx, f.metrics, f.filters, f.modifiers)
+	metas, _, partial, err = f.FetchWithMarks(ctx)
```

Comment:
We have briefly discussed this earlier in the PR. One possibility would be to ask each filter if it needs deletion marks, and only fetch them if there is any such filter. But we concluded that we should always fetch deletion marks.

In the future, we could use these fetched marks in compactor, instead of relying on `ignoreDeletionMarkFilter`.